### ISSUE TITLE
Notes controller calls set_locale after lookup_user which can cause unexpected language for 404 users notes

### ISSUE THREAD

**Issue by Firefishy** [openstreetmap/openstreetmap-website#5251]
### URL

https://www.openstreetmap.org/user/Omaraltab/notes

### How to reproduce the issue?

Refreshing the URL occasionally returns the 404 page in an unexpected language.

https://github.com/openstreetmap/openstreetmap-website/blob/acd899d26dbf09620d598d74759fc6f7aac4846a/app/controllers/notes_controller.rb#L12

### Screenshot(s) or anything else?

Page unexpectedly in German:
<img width="1011" alt="Screenshot 2024-10-07 at 11 40 39" src="https://github.com/user-attachments/assets/9a733469-4572-45a1-8968-0ad472914c8a">

**Comment by tomhughes** [comment]
We probably need to review all the `lookup_user` calls and make sure they come after `set_locale` and probably `web_timeout` as well though that's unlikely to matter much.

**Comment by mmd-osm** [comment]
Related issue for API error messages: #2227

**Comment by tomhughes** [comment]
It's not really related at all though other than in the most tangential way.

**Comment by mmd-osm** [comment]
Right, I had another issue in mind where you would see error messages in whatever language that happens to be set previously. I can't find it at the moment...
